libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -78,8 +78,16 @@ public static void AddBotRuntime(this IServiceCollection services, IConfiguratio
         internal static void AddBotRuntimeSkills(this IServiceCollection services, IConfiguration configuration)
         {
             var skillSettings = configuration.GetSection(SkillSettings.SkillSettingsKey).Get<SkillSettings>();
+            List<SkillConfigurationEntry> settings = configuration.GetSection("skill").Get<List<SkillConfigurationEntry>>();
+            if (settings?.Count > 0)
```

Comment:
I think this may not work, based on the code, if a bot is a skill consumer, it can not be a skill (only skills I call will be allowed to call me back). 

In theory a bot can be a consumer and a skill at the same time, the AuthenticationConfiguration instance passed to CloudSkillHandler should have the list skills my bot will be calling when consuming skills and the AuthenticationConfiguration passed to CloudAdapter should have the list of parent bots that can call the bot as a skill.

Reply:
In order to allow a bot to be both a host and a skill, the shape of settings would need to look like the following:
```json
"runtimeSettings:skills": {
  "AllowedCallers": [ "host1AppId", "host2AppId" ],
  "AllowedSkills": [
    {
      "Key": "skill1",
      "AppId": "skill1AppId",
      "Endpoint": "skill1.azurewebsites.net/api/messages"
    }
  ]
}
```
The `ClaimsValidator` needs to maintain 2 lists - callers, and skills. However, I am not sure how the `ValidateClaimsAsync` method would know which list to look up to validate. @gabog - Is there a way to know this info at runtime?

Reply:
For now Composer doesn't allow a skill to call another skill, so even though this may be a scenario we want to support, it's  not possible without doing some extensive manual hacking of a Composer created bot. Having skills capable of calling other skills is a larger issue where we need to make changes to Composer and the runtime to support this capability. 

The changes in the PR were only intended to plug the hole we have today in that all Claims Validation for skills for both root bots and skills are using an AllowedSkillsClaimsValidator, where we should be using an AllowedCallersClaimsValidator for root bots checking callbacks from Skills to make sure the only skills that can call back in are those the root bot is configured to call out to. The information needed by the AllowedCallersClaimsValidator can be inferred from the list of skills that the root bot is calling to, checking that any callbacks from skills have an AppId that matches one of the AppIds of a skill that the root bot calls to.

I'd recommend we push this change today, it allows us to better handle root bot scenarios and properly validates the callbacks without requiring users to configured entries into the configuration that aren't really needed. 

Since the bigger issue of allowing skills to call skills will require more work. I've documented this in Item #5738 that can be carried forward as something to consider in a future release.